templates/fine-tune-llama2/train.py

```diff
@@ -0,0 +1,871 @@
+        "--num-devices", "-nd", type=int, default=4, help="Number of devices to use."
+    )
+
+    parser.add_argument(
+        "--grad_accum", type=int, default=1, help="Gradient accumulation steps."
+    )
+    parser.add_argument(
+        "--train_path",
+         type=str,
+         default="./data/train.jsonl",
+         help="Path to training jsonl file"
+    )
+
+    parser.add_argument(
+        "--test_path", 
+        type=str, 
+        default="./data/test.jsonl",
+        help="Path to testing jsonl file"
+    )
+
+    parser.add_argument(
+        "--special_token_path", 
+        type=str, 
+        default="./data/tokens.json",
+        help="Path to token json file"
+    )
+
+    parser.add_argument(
+        "--no-grad-ckpt",
+        action="store_true",
+        help="If passed, will not use gradient checkpointing.",
+    )
+    parser.add_argument(
+        "--output_dir", 
+        type=str, 
+        default="/mnt/local_storage",
+        help="Path to output directory."
+    )
+
+    # parser.add_argument(
+    #     "--model_name", default="meta-llama/Llama-2-7b-chat-hf", type=str
+    # )
+
+    parser.add_argument(
+        "--num-epochs", 
+        type=int, 
+        default=3, 
+        help="Number of epochs to train for."
+    )
+
+    parser.add_argument(
+        "--num-checkpoints-to-keep",
+        type=int,
+        help=(
+            "Number of checkpoints to keep, if None, all checkpoints will be kept, "
+            "if set to n>=1, the top n checkpoint with min. evaluation perplexity "
+            "will be kept."
+        ),
+        default=1,
+    )
+
+    parser.add_argument(
+        "--lr", 
+        type=float, 
+        default=5e-6, 
+        help="Learning rate to use."
+    )
+
+    parser.add_argument(
+        "--ctx-len", type=int, default=512, help="Context length to use."
+    )
+
+    parser.add_argument(
+        "--as-test",
+        action="store_true",
+        help="If passed, will run the script in test mode.",
+    )
+
+    parser.add_argument(
+        "--ds-config",
+        type=str,
+        default="./deepspeed_configs/zero_3_llama_2_7b.json",
+        help="Deepspeed config json to use.",
+    )
+
+    parser.add_argument(
+        "--lora",
+        action="store_true",
+        default=False,
+        help="If passed, will enable parameter efficient fine-tuning with LoRA ("
+        "https://arxiv.org/pdf/2106.09685.pdf).",
+    )
+
+    parser.add_argument(
+        "--size",
+        default="7b",
+        choices=["7b", "13b", "70b"],
+        help="Model size. Choose between 7b, 13b, and 70b",
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    print("Fine-tuning model...")
+    args = parse_args()
+
+    # Create the config with args for training.
+    config = vars(args)
+
+    SIZE = "7B" # Default model size
+    LR = 5e-6 # Default learning rate for full-parameter fine-tuning
+
+    # Process arguments
+    for arg in sys.argv[1:]:
```

Comment:
Let's just use args here instead of sys.argv?
argparse nicely encapsulates this functionality.
You can just do `args.size`.

Reply:
Done.